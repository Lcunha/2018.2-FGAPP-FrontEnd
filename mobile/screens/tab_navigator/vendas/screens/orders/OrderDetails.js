import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    Image,
    ScrollView,
    CardItem,
    Button,
    Alert,
    ImageBackground,
} from 'react-native';
import styles from '../../components/styles'
import jwt_decode from 'jwt-decode'
import { LinearGradient } from 'expo';

class OrderDetails extends Component {

    _onPressButton = async () => {
      Alert.alert("O botao foi apertado. ");
    }

    loadUser(token, order){
      const get_product_path = `${process.env.VENDAS_API}/api/get_name/`;

      var name = 'UsuÃ¡rio sem nome';
  		fetch(get_product_path, {
  			method: 'POST',
  			headers: {
  			'Content-Type': 'application/json',
  			},
  			body: JSON.stringify({
  				'token': token,
        			 'user_id': order.fk_buyer,
  			}),
  		})
  			.then((response) => { return response.json() })
  			.then((responseJson) => {
          if(responseJson.photo != '')
            name=responseJson.name;
  			})
  			.catch((error) => {
  				console.error(error);
        });

      return name;
    }

    loadProduct(token, order){
      const get_product_path = `${process.env.VENDAS_API}/api/get_product/`;

      var photo = 'http://www.piniswiss.com/wp-content/uploads/2013/05/image-not-found-4a963b95bf081c3ea02923dceaeb3f8085e1a654fc54840aac61a57a60903fef-300x199.png';
  		fetch(get_product_path, {
  			method: 'POST',
  			headers: {
  				'Content-Type': 'application/json',
  			},
  			body: JSON.stringify({
  				'token': token,
          'product_id': order.fk_product,
  			}),
  		})
  			.then((response) => { return response.json() })
  			.then((responseJson) => {
          console.log(responseJson.photo);
          if(responseJson.photo != '')
            photo=responseJson.photo;
  			})
  			.catch((error) => {
  				console.error(error);
        });

      return photo;
    }

    render() {
      const {state} = this.props.navigation;
      var token = state.params ? state.params.token : undefined;
      var order = state.params ? state.params.order : undefined;
      var user = jwt_decode(token);
      var photo = this.loadProduct(token, order);
      var name = this.loadUser(token, order);

        return (
          <ScrollView>
            <View style={styles.details_main}>
              <Image
                style={styles.image }
                source={{ uri: photo}}
              >
              </Image>
              <View style={{ flexDirection: 'row', paddingBottom: 2, paddingLeft: 10}}>
                <View style={{flexDirection: 'column', width: '70%',}}>
                  <Text style={{fontSize: 25}}>{order.product_name}</Text>
                  <Text style={{fontSize: 20}}>R$ {parseFloat(order.total_price).toFixed(2)}</Text>
                </View>
                <View style={{flexDirection: 'column', width: '30%'}}>
                  <Text style={{fontSize: 12}}>Quantidade</Text>
                  <Text style={{fontSize: 16}}>{order.quantity}</Text>
                </View>
              </View>
              <View style={{ flexDirection: 'column', paddingLeft: 10 }}>
                <Text style={{fontSize: 20}}>Cliente: {name}</Text>
                <View style={{height: 10}}/>
                <Text style={{fontSize: 16, color: '#5A5A5A'}}>{order.buyer_message}</Text>
                <View style={{height: 50}}/>
              </View>
              <View style={{flexDirection: 'row'}}>
                <View style={{width: '70%'}}/>
                <Button
                  color="#0EAC6F"
                  title="Atendido"
                  onPress={this._onPressButton}
                />
              </View>
            <View style={{padding:20}}/>
            </View>
          </ScrollView>
        );
    }
}
export default OrderDetails;

const styless = StyleSheet.create({
    container: {
        flex: 1,
        //width: '100%',
    }
});
