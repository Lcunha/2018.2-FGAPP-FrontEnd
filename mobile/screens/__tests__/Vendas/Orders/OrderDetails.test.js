import React from 'react';
import OrderDetails from '../../../tab_navigator/vendas/screens/orders/OrderDetails';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import fetchMock from 'fetch-mock';

Enzyme.configure({adapter: new Adapter()});

describe('Testing navigation', () => {

    let wrapper = null
    const spyNavigate = jest.fn()
    const props = {
        navigation: {
            token: 'randomToken',
            navigate: spyNavigate,
            state: {
                params:{
                    order: {
                        "buyer_message": "This is a buyer message",
                        "fk_buyer": 3,
                        "fk_product": 2,
                        "id": 5,
                        "product_name": "This is a product name",
                        "quantity": 1,
                        "status": 1,
                        "total_price": 127.99,
                    },
                },
            },
        },
    }
    const params = {
        token: 'randomToken',
    }

    beforeEach(() => {
        wrapper = shallow(<OrderDetails {...props} />)
        wrapper.setState({ params: params })
    })

    it('should test navigation', async () => {
        await wrapper.instance()._goBack(params)
        expect(spyNavigate).toHaveBeenCalled()
    })
})

describe('Test _cancelButton', () => {
    const navigation = {
        state: {
            params: {
                token: "ThisIsAToken",
                order: {
                    "buyer_message": "This is a buyer message",
                    "fk_buyer": 3,
                    "fk_product": 2,
                    "id": 5,
                    "product_name": "This is a product name",
                    "quantity": 1,
                    "status": 1,
                    "total_price": 127.99,
                },
            },
        },
        navigate: jest.fn(),
    }

    const ORDER_CLOSED = 1;
    // Url to be mocked
    let set_order_status_path = '';
    let get_product_path = '';
    let get_user_path = '';

    beforeAll(() => {
        process.env.VENDAS_API = 'http://test.ip';
        set_order_status_path = `${process.env.VENDAS_API}/api/set_order_status/`;
        get_product_path = `${process.env.VENDAS_API}/api/get_product/`;
        get_user_path = `${process.env.VENDAS_API}/api/get_name/`;
    })

    beforeEach(() => {
        fetchMock.restore();
    })

    it('renders correctly', () => {
        const tree = renderer.create(<OrderDetails navigation={navigation}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Test buttons functions with sucess', async (done) => {
        const wrapper = shallow(<OrderDetails navigation={navigation}/>);

        fetchMock.post(set_order_status_path, {
            "buyer_message": "Message",
            "date": "2018-11-14T01:16:07.680903Z",
            "fk_buyer": 3,
            "fk_product": 2,
            "id": 5,
            "product_name": "Product 3",
            "quantity": 1,
            "status": 1,
            "total_price": 127.99,
        });

        fetchMock.post(get_product_path, {
            "description": "This is a description",
            "fk_vendor": 3,
            "id": 2,
            "name": "Name",
            "photo": "This is a photo url",
            "price": 127.99,
        });

        fetchMock.post(get_user_path, {
            "name": "UserName",
        });

        const state = {
            params: {
                token: "ThisIsAToken",
                order: {
                    "buyer_message": "This is a buyer message",
                    "fk_buyer": 3,
                    "fk_product": 2,
                    "id": 5,
                    "product_name": "This is a product name",
                    "quantity": 1,
                    "status": 1,
                    "total_price": 127.99,
                },
                request: '0',
            },
        };

        wrapper.setState(state);
        await wrapper.instance()._closeButton()
        await wrapper.instance()._cancelButton()
        await wrapper.instance()._buttonRequest()
        process.nextTick(() => {
            done();
        });

    });
})
