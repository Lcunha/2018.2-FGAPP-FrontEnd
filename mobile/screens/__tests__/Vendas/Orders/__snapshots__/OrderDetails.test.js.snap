// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test attendedOrder renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "flexDirection": "column",
          "margin": 0,
          "padding": 0,
          "width": "100%",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://www.piniswiss.com/wp-content/uploads/2013/05/image-not-found-4a963b95bf081c3ea02923dceaeb3f8085e1a654fc54840aac61a57a60903fef-300x199.png",
          }
        }
        style={
          Object {
            "height": 210,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "paddingBottom": 2,
            "paddingLeft": 10,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "width": "70%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 25,
              }
            }
          >
            This is a product name
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 20,
              }
            }
          >
            R$ 
            127.99
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "width": "30%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            Quantidade
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            1
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "paddingLeft": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Cliente: 
          Usu√°rio sem nome
        </Text>
        <View
          style={
            Object {
              "height": 10,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#5A5A5A",
              "fontSize": 16,
            }
          }
        >
          This is a buyer message
        </Text>
        <View
          style={
            Object {
              "height": 50,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "width": "60%",
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#0EAC6F",
              "flex": 0,
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "width": 100,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "color": "white",
                "fontSize": 16,
                "fontWeight": "300",
              }
            }
          >
             
            Atendido
             
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
